pubsub/api/QuickStart/QuickStart.cs

```diff
@@ -17,25 +17,20 @@
 using Google.Cloud.PubSub.V1;
 using System;
 using System.Diagnostics;
+using Google.Apis.Auth.OAuth2;
 
 namespace GoogleCloudSamples
 {
     public class QuickStart
     {
         public static void Main(string[] args)
         {
+            var credential = GoogleCredential.FromEnvironment();
             // Instantiates a client
-            PublisherClient publisher = PublisherClient.Create();
-
-            // Your Google Cloud Platform project ID
-            string projectId = "YOUR-PROJECT-ID";
-            // [END pubsub_quickstart]
-            Debug.Assert(projectId != "YOUR-PROJECT" + "-ID",
-                "Edit Program.cs and replace YOUR-PROJECT-ID with your project id.");
-            // [START pubsub_quickstart]
+            PublisherClient publisher = PublisherClient.Create(credential);
 
             // The name for the new topic
-            var topicName = new TopicName(projectId, "my-new-topic");
+            var topicName = new TopicName(credential.ProjectId, "my-new-topic");

```

Comment:
Why do we need to specify the projectId to create a topic name? Topic name is data entity that is provided to publisher's CreateTopic method. When you hold "publisher" instance, you already authenticated using credential that was passed in static factory method of PublisherClient. Therefore, the API call should know which project Id to work against when creating the topic.